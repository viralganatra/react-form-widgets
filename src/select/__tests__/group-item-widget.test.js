import React from 'react';
import GroupItem from '../group-item-widget';

function noop() {}

const props = {
    label: 'test',
    onClick: noop,
};

describe('GroupItemWidget', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<GroupItem {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should validate the prop types', () => {
        const wrapperInvalidLabel = () => shallow(
            <GroupItem {...props} label={null} />,
        );

        const wrapperInvalidOnClick = () => shallow(
            <GroupItem {...props} onClick="test" />,
        );

        expect(wrapperInvalidLabel).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidOnClick).toThrowErrorMatchingSnapshot();
    });

    it('should listen to the onClick event and call the prop onClick method', () => {
        const methodSpy = jest.fn();
        const wrapper = shallow(<GroupItem {...props} onClick={methodSpy} />);

        wrapper.find('li').simulate('click');

        expect(methodSpy).toHaveBeenCalled();
    });
});
