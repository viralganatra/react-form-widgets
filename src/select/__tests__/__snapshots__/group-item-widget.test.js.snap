exports[`GroupItemWidget should render correctly 1`] = `
<li
  onClick={[Function]}>
  test
</li>
`;

exports[`GroupItemWidget should validate the prop types 1`] = `
"Warning: Failed prop type: The prop \`label\` is marked as required in \`SelectMultichoiceGroupItem\`, but its value is \`null\`.
    in SelectMultichoiceGroupItem"
`;

exports[`GroupItemWidget should validate the prop types 2`] = `
"Warning: Failed prop type: Invalid prop \`onClick\` of type \`string\` supplied to \`SelectMultichoiceGroupItem\`, expected \`function\`.
    in SelectMultichoiceGroupItem"
`;
